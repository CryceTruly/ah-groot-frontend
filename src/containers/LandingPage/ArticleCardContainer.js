import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getArticles } from '../../actions/ArticleAction';
import ArticleCard from '../../components/landingPage/ArticleCard';


export class AricleCardContainer extends Component {
  componentWillMount() {
    this.props.getArticles();
  }

  render() {
      return(
    <div className="row">
    <div className="col s12">Recent articles</div>
    <div className="col s6">
    {
    (this.props.articles.length > 1) ?
    this.props.articles.slice(0,3).map(element => {
      return (
            <ArticleCard
            key={element.slug}
            title = {element.title}
            description = {element.description}
            body={element.body}
            reading_time={element.reading_time}
            updated_at={element.updated_at}
            />
      )
    }): <span></span>
    }
    </div>
    <div className="col s6">
    {
    (this.props.articles.length > 1) ?
    this.props.articles.slice(3).map(element => {
      return (
            <ArticleCard
            key={element.slug}
            title = {element.title}
            description = {element.description}
            body={element.body}
            reading_time={element.reading_time}
            updated_at={element.updated_at}
            />
      )
    }): <span></span>
    }
    </div>
  </div>
      );
  }
}

const mapStateToProps = state => {
return {
  articles: state.articles,
  categories: state.categories
}};

const mapDispatchToProps = dispatch => {
  return {
    getArticles: function() {
      dispatch(getArticles());
    },
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AricleCardContainer);
