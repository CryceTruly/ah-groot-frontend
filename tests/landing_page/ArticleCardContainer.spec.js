import configureStore from 'redux-mock-store';
import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from "redux-thunk";
import { Provider } from 'react-redux';
import moxios from 'moxios';
import data from './maxios_mock';

import ArticleCardContainer, { ArticleCardContainer as App} from '../../src/containers/LandingPage/ArticleCardContainer';

describe('ArticleCardContainer', () => {
  const initialState = {
    articles : data.article.articles.results
  };
  const mockStore = configureStore([thunk]);
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
    moxios.install();
  });

  afterEach(function () {
    moxios.uninstall();
  });

  it('should render without crashing', () => {
    const wrapper = shallow( <Provider store={store}><ArticleCardContainer /></Provider> );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without crashing Again', () => {
    const wrapper = mount( <Provider store={store}><ArticleCardContainer /></Provider> );
  });

  it('should render when state is updated', () => {
    const wrapper = mount( <Provider store={store}><ArticleCardContainer /></Provider> );
    wrapper.setProps({ articles: data.article.articles.results });
    wrapper.update();
    expect(wrapper.prop('articles').length).toEqual(6);
  });

  it('should render empty span when there are no props', () => {
    const wrapper = mount( <Provider store={store}><ArticleCardContainer /></Provider> );
    wrapper.setProps({ articles: [] });
    wrapper.update();
    expect(wrapper.prop('articles').length).toEqual(0);
  });
});
