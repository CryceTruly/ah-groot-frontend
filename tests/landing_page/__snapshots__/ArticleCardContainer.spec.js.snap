// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCardContainer should render without crashing 1`] = `ShallowWrapper {}`;
